source/MaterialXView/Material.cpp

```diff
@@ -401,17 +417,20 @@ bool Material::bindImage(const mx::FilePath& filename, const std::string& unifor
     if (!imageHandler->acquireImage(resolvedFilename, desc, true, fallbackColor) && !filename.isEmpty())
     {
         std::cerr << "Failed to load image: " << resolvedFilename.asString() << std::endl;
+        return returnPath;
     }
 
     // Bind the image and set its sampling properties.
-    int textureLocation = imageHandler->getBoundTextureLocation(desc.resourceId);
-    if (textureLocation < 0)
-        return false;
-
-    _glShader->setUniform(uniformName, textureLocation, false);
-    imageHandler->bindImage(resolvedFilename, samplingProperties);
-
-    return true;
+    if (imageHandler->bindImage(resolvedFilename, samplingProperties))
```

Comment:
This was incorrect. It was testing if it was bound and failing if not before there was an attempt to bind. This was because it was relying on acquire to occur immediately before bind and storing bound locations (which does not occur anymore). Now we try to bind. If success keep track of bound images.